examples/experimental/potree/src/potree-tile-3d-layer.ts

```diff
@@ -0,0 +1,79 @@
+import {Tile3DLayer, Tile3DLayerProps} from '@deck.gl/geo-layers';
+import {UpdateParameters} from '@deck.gl/core/typed';
+import { Viewport } from '@deck.gl/core';
+import {Source} from '@loaders.gl/loader-utils';
+import { PointcloudTileset } from './pointcloud-tileset';
+
+export type PotreeTile3DLayerProps = {
+  dataSource: Source
+}
+
+export default class PotreeTile3DLayer<
+  DataT = any,
+  // eslint-disable-next-line @typescript-eslint/ban-types
+  ExtraProps extends {} = {}
+> extends Tile3DLayer<DataT, Tile3DLayerProps & PotreeTile3DLayerProps & ExtraProps > {
+  static layerName = 'PotreeTile3DLayer';
+  static defaultProps = Tile3DLayer.defaultProps as any;
+
+  //@ts-expect-error
+  updateState({props, oldProps, changeFlags}: UpdateParameters<this>): void {
+    if (props.data && props.data !== oldProps.data) {
+      // eslint-disable-next-line @typescript-eslint/no-floating-promises
+      this._loadTileset(props.data);
+    }
+
+    if (changeFlags.viewportChanged) {
+      const {activeViewports} = this.state;
+      const viewportsNumber = Object.keys(activeViewports).length;
+      if (viewportsNumber) {
+        this._updateTileset(activeViewports);
+        this.state.lastUpdatedViewports = activeViewports;
+        this.state.activeViewports = {};
+      }
+    }
+    if (changeFlags.propsChanged) {
+      const {layerMap} = this.state;
+      for (const key in layerMap) {
+        layerMap[key].needsUpdate = true;
+      }
+    }
+  }
+
+  private async _loadTileset(tilesetUrl) {
+    const {loadOptions = {}} = this.props;
+
+    const dataSource = this.props.dataSource.createDataSource(tilesetUrl, {});
```

Comment:
Hmm - should we keep a reference to the dataSource and call destroy() on it in the end? Not sure dataSources have destroy() methods yet.

Maybe we should also add a `destroy()` method to the PointCloudTIleset as well? Or just rely on garbage collection?